vllm/model_executor/models/chatglm.py

```diff
@@ -0,0 +1,338 @@
+# coding=utf-8
+# Adapted from
+# https://huggingface.co/THUDM/chatglm3-6b/blob/main/modeling_chatglm.py
+# LICENSE: https://huggingface.co/THUDM/chatglm3-6b/blob/main/MODEL_LICENSE
+"""Inference-only ChatGLM model compatible with HuggingFace weights.
+
+The input of the model is flattened to a 1D tensor of tokens. The model uses
+InputMetadata to extract the original 2D shape of the input.
+"""
+from typing import List, Optional, Tuple
+
+import torch
+from torch import nn
+
+from vllm.model_executor import InputMetadata
+from vllm.model_executor.layers.activation import SiluAndMul
+from vllm.model_executor.layers.attention import PagedAttentionWithRoPE
+from vllm.model_executor.layers.layernorm import RMSNorm
+from vllm.model_executor.layers.sampler import Sampler
+from vllm.model_executor.parallel_utils.layers import ColumnParallelLinear, RowParallelLinear, VocabParallelEmbedding
+from vllm.model_executor.parallel_utils.parallel_state import get_tensor_model_parallel_rank
+from vllm.model_executor.parallel_utils.parallel_state import get_tensor_model_parallel_world_size
+from vllm.model_executor.weight_utils import hf_model_weights_iterator, load_tensor_parallel_weights, \
+    load_padded_tensor_parallel_vocab
```

Comment:
Please use `()` for line continuation:
```suggestion
from vllm.model_executor.weight_utils import (
    hf_model_weights_iterator, load_tensor_parallel_weights, 
    load_padded_tensor_parallel_vocab)
```